src/interface/params.hpp

```diff
@@ -39,11 +39,21 @@ class Params {
   /// Throws an error if the key is already in use
   template <typename T>
   void Add(const std::string &key, T value) {
-    PARTHENON_REQUIRE_THROWS(!(hasKey(key)), "Key " + key + "already exists");
+    PARTHENON_REQUIRE_THROWS(!(hasKey(key)), "Key " + key + " already exists");
     myParams_[key] = std::unique_ptr<Params::base_t>(new object_t<T>(value));
     myTypes_[key] = std::string(typeid(value).name());
   }
 
+  /// Updates existing object
+  /// Throws an error if the key is not already in use
+  template <typename T>
+  void Update(const std::string &key, T value) {
+    PARTHENON_REQUIRE_THROWS((hasKey(key)), "Key " + key + "missing.");
+    PARTHENON_REQUIRE_THROWS(!(myTypes_.at(key).compare(std::string(typeid(T).name()))),
+                             "WRONG TYPE FOR KEY '" + key + "'");
```

Comment:
Hmm, I guess we can't use `std::is_same` or any other compile-time check here because we don't have the type of the existing key at compile time, right?

Reply:
Yeah it's a runtime thing.